Std/Tactic/TryThis.lean

```diff
@@ -523,3 +524,57 @@ def addTermSuggestions (ref : Syntax) (es : Array Expr)
     (codeActionPrefix? : Option String := none) : MetaM Unit := do
   addSuggestions ref (← es.mapM delabToRefinableSuggestion)
     (origSpan? := origSpan?) (header := header) (codeActionPrefix? := codeActionPrefix?)
+
+open Lean Elab Elab.Tactic PrettyPrinter Meta Std.Tactic.TryThis
+
+/-- Add a suggestion for `have : t := e`. -/
+def addHaveSuggestion (ref : Syntax) (t? : Option Expr) (e : Expr)
+  (origSpan? : Option Syntax := none) : TermElabM Unit := do
+  let estx ← delabToRefinableSyntax e
+  let prop ← isProp (← inferType e)
+  let tac ← if let some t := t? then
+    let tstx ← delabToRefinableSyntax t
+    if prop then
+      `(tactic| have : $tstx := $estx)
+    else
+      `(tactic| let this : $tstx := $estx)
```

Comment:
does this put `this†` in the output?

Reply:
Indeed, it does, and there's a FIXME in Mathlib's `test/propose.lean` about this.

Reply:
it should use ``$(mkIdent `this)`` instead

Reply:
Okay, I've fixed the behaviour of Mathlib's `have?` and `observe?` in https://github.com/leanprover-community/mathlib4/pull/9454, and will modify this PR accordingly.